app.rb

```diff
@@ -0,0 +1,232 @@
+require_relative 'teacher'
+require 'date'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    puts "Welcome to School Library App!\n\n"
+  end
+
+  def run
+    loop do
+      puts main_section
+      choice = get_number('', 'Please enter a valid number!')
+      return if choice == 7
+
+      run_available_options(choice)
+    end
+  end
+
+  private
+
+  def run_available_options(choice)
+    case choice
+    when 1
+      list_books(@books)
+    when 2
+      list_people(@people)
+    when 3
+      add_person
+    when 4
+      @books << add_book
+    when 5
+      add_rental(@books, @people)
+    when 6
+      list_rentals(@people)
+    else
+      puts ''
+    end
+  end
+
+  def main_section
+    'Please choose an option by entering a number:
+1 - List all books
+2 - List all people
+3 - Create a person
+4 - Create a book
+5 - Create a rental
+6 - List all rentals for a given person
+7 - Exit'
+  end
+
+  def add_person
+    person_type = get_number('Do you want to create a Student (1) or a teacher (2)?. Back to main (any number): ',
+                             'Please enter a valid number!')
+    case person_type
+    when 1
+      @people << add_student
+    when 2
+      @people << add_teacher
+    else
+      puts ''
+    end
+  end
+
+  def add_student
+    age = get_number('Age: ', 'Please enter a valid number!')
+    name = get_str('Name: ', 'Please enter a proper name!')
+    permission = get_bool('Parent permission (y/n): ', 'Please enter either (y/n)!')
+    student = Student.new(age, name, parent_permission: permission)
+    return unless student
+
+    puts "Student #{name} is added!\n\n"
+    student
+  end
+
+  def add_teacher
+    specialization = get_str('Specialization: ', 'Please enter a proper specialization!')
+    age = get_number('Age: ', 'Please enter a valid number!')
+    name = get_str('Name: ', 'Please enter a proper name!')
+    teacher = Teacher.new(specialization, age, name)
+    return unless teacher
+
+    puts "Teacher #{name} is added!\n\n"
+    teacher
+  end
+
+  def add_book
+    title = get_str('Title: ', 'Please enter a valid title!')
+    author = get_str('Author: ', 'Please enter a valid author!')
+    book = Book.new(title, author)
+    return unless book
+
+    puts "Book #{title} is created successfully!\n\n"
+    book
+  end
+
+  def list_books(books, with_id: false)
+    if with_id
+      books.each_with_index { |book, idx| puts "#{idx}) Title: #{book.title}, Author: #{book.author}" }
+    else
+      books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+    puts ''
+  end
+
+  def list_people(people, with_id: false)
+    if with_id
+      people.each_with_index do |person, idx|
+        puts "#{idx}) [#{person.class}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+      end
+    else
+      people.each { |person| puts "[#{person.class}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+    end
+    puts ''
+  end
```

Comment:
[OPTIONAL] When the user selects option 1 or 2 before creating a book or person, nothing happens. Which is not good for user experience. Just as you have done for the rentals, kindly consider adding a feedback message like `"No books added yet"` or `"No person created yet"` to give a good user experience and a better feedback . :sweat_smile: